src/pubsub.js

```diff
@@ -43,6 +43,8 @@ module.exports = (node) => {
       if (floodSub.listenerCount(topic) === 0) {
         floodSub.unsubscribe(topic)
       }
+
+      setImmediate(() => callback())
```

Comment:
Since the callback is being added, even though it should have already exists, we should check that it is actually a function before invoking it.

Reply:
makes sense, I added a check